### PR TITLE
Create a PowerCLI sample.

### LINKED ISSUE #40: PowerCLI sample
We should make a sample that uses a containerized PowerCLI to script against the API.

[This article](https://blogs.vmware.com/PowerCLI/2013/03/back-to-basics-connecting-to-vcenter-or-a-vsphere-host.html) makes me think we should have our binding setting up the equivalent of: `%APPDATA%\VMware\credstore\vicredentials.xml`

### PR DESCRIPTION
**Pull request by mattmoor**
This doesn't reflect the target experience, but it works.  Use this as a conversation starter and iterate.

Fixes: https://github.com/vmware-tanzu/sources-for-knative/issues/40

### CODE REVIEW

File: samples/powercli/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+## `VSphereBinding` with `PowerCLI`
+
+This sample builds on our [previous sample](../govc/README.md) to show how
+to use `VSphereBinding` to authenticate `PowerCLI` (a more familiar tool
+to most admins).
```
Comment (mattmoor) [inline samples/powercli/README.md:5]: Format markdown:
```suggestion
This sample builds on our [previous sample](../govc/README.md) to show how to
use `VSphereBinding` to authenticate `PowerCLI` (a more familiar tool to most
admins).
```

File: samples/powercli/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+We are going to make use of the `vmware/powerclicore` container image in
+this sample.
```
Comment (mattmoor) [inline samples/powercli/README.md:8]: Format markdown:
```suggestion
We are going to make use of the `vmware/powerclicore` container image in this
sample.
```

File: samples/powercli/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+### Pre-requisites
+
+**Unlike the previous examples**, this example does not work with `vcsim`, so
+you will need a real vSphere environment set up, with credentials in a Secret
+named `vsphere-credentials`.
+
+
```
Comment (mattmoor) [inline samples/powercli/README.md:16]: Format markdown:
```suggestion

```

File: samples/powercli/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+Once you have your binding ready, apply it with:
+
+```shell
+kubectl apply -f binding.yaml
+```
+
+### Script against vSphere with `PowerCLI`
+
+We are going to run the following Job to script some automation using `PowerCLI`:
```
Comment (mattmoor) [inline samples/powercli/README.md:50]: Format markdown:
```suggestion
We are going to run the following Job to script some automation using
`PowerCLI`:
```

File: samples/powercli/README.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      restartPolicy: Never
+      containers:
+      - name: dump-events
+        image: docker.io/vmware/powerclicore
+        command: ["pwsh", "-Command"]
+        args:
+        - |
+          # Log into the VI Server
+          Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Confirm:$false | Out-Null
+          Connect-VIServer -Server ([System.Uri]$env:GOVC_URL).Host -User $env:GOVC_USERNAME -Password $env:GOVC_PASSWORD
```
Comment (mattmoor) [inline samples/powercli/README.md:76]: Format markdown:
```suggestion
        - name: dump-events
          image: docker.io/vmware/powerclicore
          command: ["pwsh", "-Command"]
          args:
            - |
              # Log into the VI Server
              Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Confirm:$false | Out-Null
              Connect-VIServer -Server ([System.Uri]$env:GOVC_URL).Host -User $env:GOVC_USERNAME -Password $env:GOVC_PASSWORD
```

File: samples/powercli/README.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+          # Get Events and write them out.
+          Get-VIEvent | Write-Host
```
Comment (mattmoor) [inline samples/powercli/README.md:79]: Format markdown:
```suggestion
              # Get Events and write them out.
              Get-VIEvent | Write-Host
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `prettier --write --prose-wrap=always $(find -name '*.md' | grep -v vendor | grep -v .github | grep -v docs/cmd/)`

File: samples/powercli/README.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+This Job authenticates `PowerCLI` using our injected credentials, and dumps the event logs.  You can run it with:
```
Comment (mattmoor) [inline samples/powercli/README.md:82]: Format markdown:
```suggestion
This Job authenticates `PowerCLI` using our injected credentials, and dumps the
event logs. You can run it with:
```

**Comment by mattmoor** [comment]
I'd love to get an e2e test for this, but I couldn't get it working against vcsim...

File: samples/powercli/README.md (hunk 1 above)
Comment (mattmoor) [inline samples/powercli/README.md:5]: Format markdown:
```suggestion
This sample builds on our [previous sample](../govc/README.md) to show how to
use `VSphereBinding` to authenticate `PowerCLI` (a more familiar tool to most
admins).
```

File: samples/powercli/README.md (hunk 2 above)
Comment (mattmoor) [inline samples/powercli/README.md:8]: Format markdown:
```suggestion
We are going to make use of the `vmware/powerclicore` container image in this
sample.
```

File: samples/powercli/README.md (hunk 3 above)
Comment (mattmoor) [inline samples/powercli/README.md:16]: Format markdown:
```suggestion

```

File: samples/powercli/README.md (hunk 4 above)
Comment (mattmoor) [inline samples/powercli/README.md:50]: Format markdown:
```suggestion
We are going to run the following Job to script some automation using
`PowerCLI`:
```

File: samples/powercli/README.md (hunk 5 above)
Comment (mattmoor) [inline samples/powercli/README.md:76]: Format markdown:
```suggestion
        - name: dump-events
          image: docker.io/vmware/powerclicore
          command: ["pwsh", "-Command"]
          args:
            - |
              # Log into the VI Server
              Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Confirm:$false | Out-Null
              Connect-VIServer -Server ([System.Uri]$env:GOVC_URL).Host -User $env:GOVC_USERNAME -Password $env:GOVC_PASSWORD
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `prettier --write --prose-wrap=always $(find -name '*.md' | grep -v vendor | grep -v .github | grep -v docs/cmd/)`

File: samples/powercli/README.md (hunk 6 above)
Comment (mattmoor) [inline samples/powercli/README.md:79]: Format markdown:
```suggestion
              # Get Events and write them out.
              Get-VIEvent | Write-Host
```

File: samples/powercli/README.md (hunk 7 above)
Comment (mattmoor) [inline samples/powercli/README.md:82]: Format markdown:
```suggestion
This Job authenticates `PowerCLI` using our injected credentials, and dumps the
event logs. You can run it with:
```

File: samples/powercli/binding.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  name: vsphere-jobs
+spec:
+  subject:
+    apiVersion: batch/v1
+    kind: Job
+    selector:
+      matchLabels:
+        role: vsphere-job
+
+  address: REPLACE_ME
```
Comment (vaikas) [inline samples/powercli/binding.yaml:13]: do you want a comment here also about needing to point to real Vsphere? Could use a yaml comment?

File: samples/powercli/job.yaml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+      restartPolicy: Never
+      containers:
+      - name: dump-events
+        image: docker.io/vmware/powerclicore
+        command: ["pwsh", "-Command"]
+        args:
+        - |
+          # Log into the VI Server
+          Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Confirm:$false | Out-Null
+          Connect-VIServer -Server ([System.Uri]$env:GOVC_URL).Host -User $env:GOVC_USERNAME -Password $env:GOVC_PASSWORD
```
Comment (vaikas) [inline samples/powercli/job.yaml:24]: could the problem with the sim not working be that you don't have the GOVC_INSURE env variable specified here?
GOVC_INSECURE=1
I assume the -InvalidCertificateAction ignore would do the same, but just curious if you tried that as well?

Comment (mattmoor) [reply samples/powercli/job.yaml:24]: Yeah, I don't have this keying off of `GOVC_INSECURE`, but the line above is what that *should* guard.

File: samples/powercli/binding.yaml (hunk 8 above)
Comment (mattmoor) [reply samples/powercli/binding.yaml:13]: done

**Review by vaikas — APPROVED** [review]

**Pull request merged**